src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+### Service Schema Description
+
+Exonum provides a lot of stored named collections for the service data. The main
+collection types are sets, lists and maps. Data organization inside the
+collections is arranged in two ways - ordinary collections and [Merkelized
+collections](https://en.wikipedia.org/wiki/Merkle_tree); the latter allow to
+provide cryptographic evidence of the authenticity of the data to the clients of
+the system (for example, that an element is stored in the collection under a
+certain key). The [blockchain state](../glossary.md#blockchain-state) is
+influenced only by the Merkelized collections.
+
+For the detailed description of all Exonum collection types see the relevant
+[documentation](../architecture/storage/#table-types). In Java,
+implementations of the collections are located in
+`com.exonum.binding.storage.indices` package. This package documentation
+describes the peculiarities of their use.
+
+!!! note
+    `SparseListIndex` is not yet supported in Java. Let us know if it may be
+    useful for you!
+
+Collections work with a database view - either `Snapshot`, which is read-only;
+or `Fork`, which is mutable and allows performing modifying operations. The
+database representation is provided by the framework: `Snapshot` can be
+requested at any time, while `Fork` - only when the transaction is executed. The
+lifetime of these objects is limited by the scope of the method to which they
+are transferred.
+
+Exonum stores arrays of bytes in the storage collections. Therefore, the user
+must implement serialization of values ​​stored in the collection. Java Binding
+provides *serializers* for standard and some commonly used types, see
+`com.exonum.binding.storage.serialization.StandardSerializers`.
+
+!!! note "Example of ProofMapIndex Creation"
+    ```java
+    void updateBalance(Fork fork) {
+      var name = "balanceById";
+      var balanceById = ProofMapIndexProxy.newInstance(name, fork, StandardSerializers.hash(),
+          StandardSerializers.longs());
+      balanceById.put(id, newBalance);  
+    }
+    ```
+
+A set of named collections constitute a *service scheme*. For convenient access
+to service collections you can implement a factory of service collections.
+
+*The state of the service in the blockchain* is determined by the list of the
+root hashes of its Merkelized collections. When using `AbstractService`, the
+root hash list must be defined in the schema class that implements
+`com.exonum.binding.service.Schema` interface; when implementing `Service`
+directly - in the service itself.
+
+### Service Transactions Description
+
+Exonum transactions allow you to perform modifying atomic operations with the
+storage. Transactions are executed sequentially, in the order determined by the
+consensus of the nodes in the network.
+
+#### Messages
+
+Transactions are transmitted by the service client as an Exonum message. The
+transaction data in the message can be serialized using an arbitrary algorithm
```

Comment:
I'd rather say "the transaction payload"; if we consider transactions as a subtype of messages, some data in the transaction (e.g., `service_id`) must be serialized in a standard way.